src/performance_stats.cpp

```diff
@@ -0,0 +1,106 @@
+/*****************************************************************************
+ *
+ * This file is part of Mapnik (c++ mapping toolkit)
+ *
+ * Copyright (C) 2017 Artem Pavlenko
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ *****************************************************************************/
+
+#include <mapnik/performance_stats.hpp>
+
+namespace mapnik {
+
+template class MAPNIK_DECL singleton<timer_stats, CreateStatic>;
+
+void timer_stats::add(std::string const& metric_name, double cpu_elapsed, double wall_clock_elapsed)
```

Comment:
Actually the time recorded does not depend on the order of evaluation of the parameters, because of the call to `timer::stop();` before adding the metrics to the singleton `timer_stats`:

https://github.com/CartoDB/mapnik/blob/53e17ac7479470eaa97b21c42f28c2965db8fc93/src/performance_stats.cpp#L92

https://github.com/mapnik/mapnik/blob/df0bbe404cf3952081cc512da6c1e3e2726a46ae/include/mapnik/timer.hpp#L82-L109